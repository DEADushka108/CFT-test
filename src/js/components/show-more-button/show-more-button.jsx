import React from 'react';
import PropTypes from 'prop-types';
import {postDetails} from '../../types/post';
import {ActionCreator} from '../../store/posts/posts';
import {connect} from 'react-redux';

const ShowMoreButton = (props) => {
  const {list, count, onClick} = props;

  return (count < list.length) ? <button className="journal__button" type="button" onClick={onClick}>Show more</button> : null;
};

ShowMoreButton.propTypes = {
  list: PropTypes.arrayOf(postDetails),
  count: PropTypes.number.isRequired,
  onClick: PropTypes.func.isRequired,
};

const mapDispatchToProps = (dispatch) => ({
  onClick() {
    dispatch(ActionCreator.setPostsCount());
  },
});

export {ShowMoreButton};
export default connect(null, mapDispatchToProps)(ShowMoreButton);

