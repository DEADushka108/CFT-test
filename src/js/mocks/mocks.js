export * from './const/const';
export * from './posts/posts';
export * from './store/store';
export * from './users/users';
export * from './comments/comments';
