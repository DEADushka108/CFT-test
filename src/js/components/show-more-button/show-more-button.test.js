import React from 'react';
import {MAX_POSTS_COUNT, noop, postsList} from '../../mocks/mocks';
import renderer from 'react-test-renderer';
import {ShowMoreButton} from './show-more-button.jsx';

it(`Should render ShowMoreButton component correctly`, () => {
  const tree = renderer
    .create(
        <ShowMoreButton
          list={postsList}
          count={MAX_POSTS_COUNT}
          onClick={noop}
        />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
