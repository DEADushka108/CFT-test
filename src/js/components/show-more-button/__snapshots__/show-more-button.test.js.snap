// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ShowMoreButton component correctly 1`] = `
<button
  className="journal__button"
  onClick={[Function]}
  type="button"
>
  Show more
</button>
`;
